src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -12176,6 +12178,38 @@ instruct cmpFastUnlock(flagsRegCR0 crx, iRegPdst oop, iRegPdst box, iRegPdst tmp
   ins_pipe(pipe_class_compare);
 %}
 
+instruct cmpFastLockLightweight(flagsRegCR0 crx, iRegPdst oop, iRegPdst box, iRegPdst tmp1, iRegPdst tmp2) %{
+  predicate(LockingMode == LM_LIGHTWEIGHT);
+  match(Set crx (FastLock oop box));
+  effect(TEMP tmp1, TEMP tmp2);
+
+  format %{ "FASTLOCK  $oop, $box, $tmp1, $tmp2" %}
+  ins_encode %{
+    __ fast_lock_lightweight($crx$$CondRegister, $oop$$Register, $box$$Register,
+                             $tmp1$$Register, $tmp2$$Register, /*tmp3*/ R0);
```

Comment:
You could avoid duplicating the nodes and distinguish the locking modes here if the register usage is the same.

Reply:
That's ok for now. We can clean everything up when removing the legacy locking. Unfortunately, `BoxLockNode` is not easy to remove completely because they are used for analyzing properties of the graph. That would need a redesign. But at least, we could implement them as empty nodes such that they don't emit any code.